pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -538,22 +562,87 @@ private static void setupTaskReceiver(Config cfg, String projectId) {
         }
     }
 
-    private static void setupFileScanResultReceiver(Config cfg, String projectId) {
+    private static void setupFileUploadListener(Config fileUploadConfig,
+            GoogleCredentials credentials,
+            String projectId,
+            PubSubConnectionManager pubsubManager) {
+        final boolean enabled = ConfigUtil.getBoolOrElse(fileUploadConfig, ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER_KEY, false);
+        if (!enabled) {
+            log.info("File upload handler is disabled ({} == false)", ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER_KEY);
+            return;
+        }
+
+        final var configSubscriptionNamePath = ConfigFile.FileUploads.UPLOADS_NOTIFICATION_SUBSCRIPTION_KEY;
+
+        if (!fileUploadConfig.hasPath(projectId)) {
+            log.info("A value for the key {} was not specified, file upload notifications will not be tracked.",
+                    ConfigFile.FileUploads.UPLOADS_NOTIFICATION_SUBSCRIPTION);
+            return;
+        }
+
+        final var configSubscriptionName = ProjectSubscriptionName.of(projectId, fileUploadConfig.getString(configSubscriptionNamePath));
+        final var uploadsBucketName = fileUploadConfig.getString(ConfigFile.FileUploads.UPLOADS_BUCKET_KEY);
+        final var storageClient = new GoogleBucketClient(projectId, credentials);
+        final var uploadsNotificationsReceiver = new UploadsEventReceiver(configSubscriptionName);
+
+        // If a topic name has been specified, attempt to create the topic,
+        // and then set up the necessary notifications on the Uploads bucket.
+        //
+        // This is an optional operation, and can be skipped by not specifing a topic name.
+        //
+        // If the credentialed account does not have sufficient privileges for this operation,
+        // an error will be thrown.
+        final var configTopicNamePath = ConfigFile.FileUploads.UPLOADS_NOTIFICATION_TOPIC_KEY;
+        ConfigUtil.getString(fileUploadConfig, configTopicNamePath)
+                .map(name -> ProjectTopicName.of(projectId, name))
+                .ifPresentOrElse(topicName -> {
+                    pubsubManager.createTopicIfNotExists(topicName);
+                    storageClient.enableNotifications(uploadsBucketName, topicName, uploadsNotificationsReceiver.acceptedEventTypes());
+
```

Comment:
If the housekeeping credentials do not have sufficient privileges to create the topics and enable notifications, this will result in an authentication error from google. This should be considered expected behavior. If topic & notification creation is not desired, the `fileUploads.uploadsNotificationTopic` key in the configuration may be omitted or `null`ed out.